source/server/http/health_check.cc

```diff
@@ -155,6 +164,32 @@ void HealthCheckFilter::onComplete() {
     Http::Code final_status = Http::Code::OK;
     if (cache_manager_) {
       final_status = cache_manager_->getCachedResponseCode();
+    } else if (cluster_min_healthy_percentages_ != nullptr &&
+               !cluster_min_healthy_percentages_->empty()) {
+      const auto clusters(context_.clusterManager().clusters());
+      for (const auto& item : *cluster_min_healthy_percentages_) {
+        const std::string& cluster_name = item.first;
+        const double min_healthy_percentage = item.second;
+        auto match = clusters.find(cluster_name);
+        if (match == clusters.end()) {
+          final_status = Http::Code::ServiceUnavailable;
+          break;
+        }
+        const auto& stats = match->second.get().info()->stats();
+        const uint64_t membership_total = stats.membership_total_.value();
+        if (membership_total == 0) {
+          if (min_healthy_percentage == 0.0) {
+            continue;
+          } else {
+            final_status = Http::Code::ServiceUnavailable;
+            break;
+          }
+        }
+        if (100.0 * stats.membership_healthy_.value() < membership_total * min_healthy_percentage) {
```

Comment:
I did it that way to avoid a division operation. But, now that you mention it, g++ might implement the division as multiplication by the reciprocal. I know it does that optimization for division by an integer constant. I'll see if it does the same thing for division by a floating point constant.

Reply:
After some testing and reading, I found that:
* g++ only only converts division by a floating point literal into multiplication by its reciprocal when a special flag `-freciprocal-math` is specified. That's probably because the optimization can change the results of computations, due to the limits of floating-point precision.
* Division is still slow compared to multiplication, but it's gotten much better in recent years: the `DIVSD` instruction that g++ generates for `/ 100.0` will add [under 20 clock cycles](http://www.agner.org/optimize/instruction_tables.pdf) of latency on recent x86 processors.

So I'll just go with `/ 100.0` to improve readability.

Reply:
The conversions from int to floating point are also expensive, so the fastest implementation probably would be to do the whole check using integers only.  I.e., during config loading, we could precompute and store `threshold = uint64_t(min_healthy_percentage) * 1000000` and then implement the check as
```
if (stats.membership_healthy_.value() < membership_total * threshold / (100 * 1000000)) {
```
For now, I'll just add a TODO comment.